log.go

```diff
@@ -80,34 +98,201 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+type activeSpanKeyType struct{}
+
+var activeSpanKey = activeSpanKeyType{}
+
+type activeSpan struct {
+	isFinished bool
+}
+
+// Start starts an opentracing span with `operationName`, using
+// any Span found within `ctx` as a ChildOfRef. If no such parent could be
+// found, StartSpanFromContext creates a root (parentless) Span.
+//
+// The return value is a context.Context object built around the
+// returned Span.
+//
+// Example usage:
+//
+//    SomeFunction(ctx context.Context, ...) {
+//        ctx := log.Start(ctx, "SomeFunction")
+//        defer log.Finish(ctx)
+//        ...
+//    }
+func (el *eventLogger) Start(ctx context.Context, operationName string) context.Context {
```

Comment:
Returning a context isn't always going to work when you can't assign that context back - https://github.com/ipfs/go-ipfs/blob/master/core/coreunix/add.go#L331 for example